[website] Release 0.8 blog post and update Download page (#1868)
alibaba/fluss: website/blog/releases/0.8.md

---

> title: "Apache Fluss 0.8: Streaming Lakehouse with Iceberg/Lance"

Reviewer: Add `Announcing` in title will be more precise for an announcement, how about: `Announcing Apache Fluss 0.8: Streaming Lakehouse for AI & Analytics` ?

sidebar_label: "Announcing Apache Fluss 0.8"
authors: [giannis, jark]
date: 2025-10-30
tags: [releases]
---

![Banner](../assets/0.8/banner.jpg)

🌊 We are excited to announce the official release of **Fluss 0.8**!

This is the first ASF release for Apache Fluss (incubating), marking a significant milestone in our journey to provide a robust streaming storage platform for real-time analytics.
Over the past four months, we’ve delivered lots of improvements and new capabilities, with more than 390+ commits, across the Streaming Lakehouse ecosystem,
including: deeper integration with Apache Flink, extensive improvements in the Streaming Lakehouse with support for [Apache Iceberg](https://iceberg.apache.org/) and [Lance](https://github.com/lancedb/lance),
and the introduction of [Delta Joins](https://cwiki.apache.org/confluence/display/FLINK/FLIP-486%3A+Introduce+A+New+DeltaJoin), which redefine efficiency in stream processing.

> Apache Fluss 0.8 marks a new era of **real-time**, **unified**, and **zero-state streaming**, designed to power the next generation of data platforms, focusing on performance, scalability, and simplicity of the overall architecture.

Reviewer: According blog content, how about using specific key words to help user understand easier ?
  
`**zero-state streaming join**, **real-time multimodal AI analytics**, and **stream-batch unification**`

- **Simplified architecture**: Eliminates complex ETL pipelines between streaming systems and vector databases.

Seamless integration: combines Fluss’s high-throughput streaming engine with Lance’s efficient columnar persistence for consistent, multi-modal data management.

```yaml title='server.yaml'
datalake.format: lance
datalake.lance.warehouse: s3://<bucket>
datalake.lance.endpoint: <endpoint>
datalake.lance.allow_http: true
datalake.lance.access_key_id: <access_key_id>
datalake.lance.secret_access_key: <secret_access_key>
```

See the [LanceDB blog post](https://lancedb.com/blog/fluss-integration/) for the full integration. You also can find more detailed instructions in the [documentation](/docs/next/streaming-lakehouse/integrate-data-lakes/lance/).

## Flink 2.1

Apache Fluss is now fully compatible with **Apache Flink 2.1**, ensuring seamless integration with the latest Flink runtime and APIs.
This update strengthens Fluss’s role as a unified streaming storage layer, providing reliable performance and consistency for modern data pipelines built on Flink.

> ### Delta Join

Author: Since this is listed under `Flink 2.1`, it is implicitly an integration feature with Flink. I’ll keep the title as is for conciseness. We’ve already noted, `This release introduces support for Delta Joins with Apache Flink`, so the context should be clear.

- **Enhanced Metadata Performance**: Addressed metadata bottlenecks during mass client restarts by strengthening the server local cache and introducing asynchronous ZooKeeper operations. This reduces metadata request latency from >10 seconds to milliseconds, ensuring stable client reconnection under load.

With these foundational stability improvements, Fluss 0.8 is now production-ready for the most demanding real-time workloads, including Alibaba’s annual Double 11 global shopping festival.

## Dynamic Configuration

### Dynamic Cluster Configs

TODO: need feature documentation

### Dynamic Table Configs

TODO: need feature documentation

## Helm Charts

This release also introduced Helm Charts. With this addition, users can now deploy and manage a full Fluss cluster using [Helm](https://helm.sh/).
The Helm chart simplifies provisioning, upgrades, and scaling by packaging configuration, manifests, and dependencies into a single, versioned release.
This should help users running Fluss on Kubernetes faster, more reliably, and with easier integration into existing CI/CD and observability setups, significantly lowering the barrier for teams adopting Fluss in production.

> TODO: add documentation link

Reviewer: nit, we'd address all TODOs before release.

With this release, Fluss can continuously ingest, update, and tier data into Lance tables with guaranteed ordering and freshness,
enabling fast synchronization between streaming pipelines and downstream ML or retrieval applications.

Key benefits include:

- **Unified multi-modal data ingestion**: Stream tabular, vector, and embedding data into Lance in real time.
- **AI/ML-ready storage**: Keep feature vectors and embeddings continuously up-to-date for model training or inference.
- **Low-latency analytics and retrieval**: Fast, continuous updates enable Lance data to be immediately usable for real-time search and recommendation.
- **Simplified architecture**: Eliminates complex ETL pipelines between streaming systems and vector databases.

Seamless integration: combines Fluss’s high-throughput streaming engine with Lance’s efficient columnar persistence for consistent, multi-modal data management.

```yaml title='server.yaml'
datalake.format: lance
datalake.lance.warehouse: s3://<bucket>
datalake.lance.endpoint: <endpoint>
datalake.lance.allow_http: true
datalake.lance.access_key_id: <access_key_id>
datalake.lance.secret_access_key: <secret_access_key>
```

> See the [LanceDB blog post](https://lancedb.com/blog/fluss-integration/) for the full integration. You also can find more detailed instructions in the [Lance Lakehouse documentation](/docs/next/streaming-lakehouse/integrate-data-lakes/lance/).

Reviewer: ```suggestion
See the [LanceDB blog post](https://lancedb.com/blog/fluss-integration/) for the full integration. You can also find more detailed instructions in the [Lance Lakehouse documentation](/docs/next/streaming-lakehouse/integrate-data-lakes/lance/).
```

## Upgrade Notes

The Fluss community tries to ensure that upgrades are as seamless as possible. However, certain changes may require users to make adjustments to certain parts of the program when upgrading to version 0.8.
Please refer to the [upgrade notes](/docs/next/maintenance/operations/upgrade-notes-0.8/) for a comprehensive list of adjustments to make and issues to check during the upgrading process.

## List of contributors
The Fluss community would like to express gratitude to all the contributors who made this release possible:

> Alibaba-HZY, CaoZhen, CenterCode, CodeDrinks, David, Giannis Polyzos, Hemanth Savasere, Hongshun Wang, Jark Wu, Jensen, Junbo Wang, Kerwin, Leonard Xu, Liebing, Maggie Cao, Mahesh Sambaram, MehulBatra, Michael Koepf, Rafael Sousa, Rion Williams, Ron, Sergey Nuyanzin, SeungMin, Wang Cheng, XianmingZhou00, Xuyang, Yang Guo, Yang Wang, Yunchi Pang, ZijunZhao, Zmm, andybj0228, buvb, cxxwang, dependabot[bot], jackylee, leosanqing, naivedogger, ocean.wy, pisceslj, totalo, xiaochen, xiaozhou, xx789, yunhong, yuxia Luo

Apache Fluss is under active development. Be sure to stay updated on the project, give it a try and if you like it,

> don’t forget to give it some ❤️ via ⭐ on [GitHub](https://github.com/apache/fluss).

Reviewer: maybe the ending could be shorter like:
```suggestion
give it a try and show your support with a ⭐ on [GitHub](https://github.com/apache/fluss).
```
